### ISSUE TITLE
Genshin impact not working on Poco F5 evolution x android 14 custom rom

### ISSUE THREAD

**Issue by BlazingFlames6073** [rushiranpise/Game-Unlocker#21]
Hello, is this project still active?

I wanted to say that this doesn't seem to be working for genshin on my device(Poco F5 on custom rom). It still displays low settings as default for my phone. Low settings as default gives you lower resolution in this game. Medium settings as default is the highest android devices can currently reach from what I know.

![Screenshot_20240116-232926_Genshin Impactws](https://github.com/rushiranpise/Game-Unlocker/assets/98283925/45372ed9-9349-4277-bbd2-5cd22ea1df9e)

I can see in logs that it's trying to spoof my device as  OnePlus 8 pro though that doesn't seem to be changing anything.

`[ 2024-01-16T23:33:46.025    10583: 22791: 22791 I/LSPosed-Bridge  ] Loading legacy module io.github.rushiranpise.gameunlocker from /data/app/~~Od3zt6p35H-OIa3Heqey3A==/io.github.rushiranpise.gameunlocker-INheshJKqTzACzC8FGaatQ==/base.apk
[ 2024-01-16T23:33:46.026    10583: 22791: 22791 I/LSPosed-Bridge  ]   Loading class io.github.rushiranpise.gameunlocker.GAMEUNLOCKER
[ 2024-01-16T23:33:46.052    10583: 22791: 22791 I/LSPosed-Bridge  ] Spoofed com.miHoYo.GenshinImpact as OnePlus 8 Pro
`

I noticed nobody made a genshin issue before this so I'd like mention my discoveries trying other spoofers. Genshin seems to only look at two pieces of information to judge graphics. They are **GPU vendor** and **GPU model**. Spoofing the GPU is what seems to get you better graphics in genshin. These two settings from Gltools managed to get me medium settings as default even though I left everything else unchanged. Genshin crashes shortly after during gameplay if you use Gltools sadly. 

![Screenshot_20240116-235308_GLTools](https://github.com/rushiranpise/Game-Unlocker/assets/98283925/49238798-1024-408a-96ed-e0e727cebd8d)

Extra info for GPU vendor:

![Screenshot_20240116-235339_GLTools](https://github.com/rushiranpise/Game-Unlocker/assets/98283925/d1ed6e9e-d3de-473e-82c9-6ced4427dfbe)

Extra info for GPU model:

![Screenshot_20240116-235443_GLTools](https://github.com/rushiranpise/Game-Unlocker/assets/98283925/918f569a-d4b2-422f-bd29-8e1f335cba43)

Sorry if I'm speaking out of line. Just trying to give as much information as possible to help. I apologize if I shouldn't have mentioned a different spoofer.

**Comment by rushiranpise** [comment]
Please check with new release
https://github.com/rushiranpise/Game-Unlocker/releases/tag/6.0.0

Force Stop & Clear game app data after updating

**Comment by BlazingFlames6073** [comment]
Hello, I'm sorry for the late reply. I saw the new release soon after you released it but I didn't notice your comment here until now. I have two things to report.

1. I tried it when it released but it sadly still didn't seem to work. My default settings were still low. Unfortunately, it seems I don't have logs to share since it's been some time and I didn't save back then.

2. The game had one of it's large updates today and it seems to have fixed the issue for my device. My default settings are medium settings now. I compared with a different device and it seems fine now without a spoof. 

I apologize for troubling you. I can't even test your normal project now since my device is at higher settings by default now. If you make custom versions which lower the settings instead just for testing purposes, I'm willing to test those.
